src/main/java/io/boomerang/service/EventProcessorImpl.java

```diff
@@ -273,14 +278,34 @@ private Map<String, String> processProperties(JsonNode eventData, String workflo
         logger.error(e.toString());
       }
     }
-
-    properties.put("eventPayload", eventData.toString());
-
-    properties.forEach((k, v) -> {
-      logger.info("processProperties() - " + k + "=" + v);
+    ObjectMapper mapper = new ObjectMapper();
+    Map<String, String> payloadProperties =
+        mapper.convertValue(eventData.get("properties"), new TypeReference<Map<String, String>>() {});
+    properties.putAll(payloadProperties);
+
+    // properties.put("eventPayload", eventData.toString());
+
+    WorkflowEntity workflow = workflowService.getWorkflow(workflowId);
+
+    List<KeyValuePair> propertyList = ParameterMapper.mapToKeyValuePairList(properties);
+    Map<String, WorkflowProperty> workflowPropMap = workflow.getProperties().stream()
+        .collect(Collectors.toMap(WorkflowProperty::getKey, WorkflowProperty -> WorkflowProperty));
+    // Use default value for password-type parameter when user input value is null when executing
+    // workflow.
+    propertyList.stream().forEach(p -> {
+      if (workflowPropMap.get(p.getKey()) != null
+          && FieldType.PASSWORD.value().equals(workflowPropMap.get(p.getKey()).getType())
+          && p.getValue() == null) {
+        p.setValue(workflowPropMap.get(p.getKey()).getDefaultValue());
+      }
```

Comment:
Should the parameter resoluton around defaults be resolved in the Workflow Execution Service and not the EventProcessor